### PR TITLE
Move to stable Microsoft.CodeAnalysis.PublicApiAnalyzers version

### PR DESCRIPTION
**Pull request by dougbu**

### CODE REVIEW

File: eng/targets/CSharp.Common.targets [hunk 1]
```diff
@@ -14,21 +14,21 @@
     <!-- Public members should not use oblivious types. Not done with all nullable annotations. -->
     <NoWarn>$(NoWarn);RS0041</NoWarn>
   </PropertyGroup>
-  <ItemGroup Condition=" '$(IsImplementationProject)' == 'true' AND
+
+  <ItemGroup Condition=" Exists('$(MSBuildProjectDirectory)\PublicAPI.*.txt') AND
+      '$(IsImplementationProject)' == 'true' AND
```
Comment (dougbu) [inline eng/targets/CSharp.Common.targets:19]: New version emits errors if Public API baseline files are missing

File: eng/targets/CSharp.Common.targets [hunk 2]
```diff
@@ -14,21 +14,21 @@
...
     <Reference Include="Microsoft.CodeAnalysis.PublicApiAnalyzers" ExcludeAssets="Compile" PrivateAssets="All" />
-
-    <AdditionalFiles Include="PublicAPI.Shipped.txt"
-        Condition=" Exists('$(MSBuildProjectDirectory)\PublicAPI.Shipped.txt') " />
-    <AdditionalFiles Include="PublicAPI.Unshipped.txt"
-        Condition=" Exists('$(MSBuildProjectDirectory)\PublicAPI.Unshipped.txt') " />
-    <AdditionalFiles Include="PublicAPI.Shipped.txt"
-        Condition=" Exists('$(MSBuildProjectDirectory)\$(_TFMDirectory)\PublicAPI.Shipped.txt') " />
-    <AdditionalFiles Include="PublicAPI.Unshipped.txt"
-        Condition=" Exists('$(MSBuildProjectDirectory)\$(_TFMDirectory)\PublicAPI.Unshipped.txt') " />
```
Comment (dougbu) [inline eng/targets/CSharp.Common.targets:30]: Done in the `.targets` file for the package now

File: eng/targets/CSharp.Common.targets [hunk 3]
```diff
@@ -14,21 +14,21 @@
...
   </ItemGroup>
+  <Target Name="_CheckIgnoredPublicApiFiles"
+      BeforeTargets="Build;Restore"
+      Condition=" Exists('$(MSBuildProjectDirectory)\PublicAPI.*.txt') AND
+      !('$(IsImplementationProject)' == 'true' AND
+        '$(DotNetBuildFromSource)' != 'true' AND
+        ! $(RepoRelativeProjectDir.Contains('Tools'))) ">
+    <Warning Text="Public API baseline files ignored." />
```
Comment (dougbu) [inline eng/targets/CSharp.Common.targets:30]: just in case 😺

File: eng/targets/CSharp.Common.targets (hunk 1 above)
Comment (Pilchie) [reply eng/targets/CSharp.Common.targets:19]: Nice - would be good to revert this once we can fix all the occurrences.

**Review by Pilchie — APPROVED** [review]

File: eng/targets/CSharp.Common.targets (hunk 1 above)
Comment (dougbu) [reply eng/targets/CSharp.Common.targets:19]: Agree. If we undid this change and eliminated analyzers and `msbuild` tasks, we'd know for sure the public API is covered. Might need to invent a special `ImNotPubliclyConsumable` flag for Microsoft.Web.Xdt.Extensions and other special cases😺

First, we need to be able to analyze projects containing `*.cshtml` or `*.razor` files:
- Microsoft.AspNetCore.Authentication.AzureAD.UI
- Microsoft.AspNetCore.Authentication.AzureADB2C.UI
- Microsoft.AspNetCore.Components.Authorization
- Microsoft.AspNetCore.Identity.UI

Do we have an issue tracking this problem❔ /cc @sharwell @mavasani

Comment (Pilchie) [reply eng/targets/CSharp.Common.targets:19]: https://github.com/dotnet/roslyn/issues/47350 is tracking it, but it might need to be fixed in aspnetcore.

**Pull request merged**